### PR TITLE
suricata: add new package

### PR DESCRIPTION
**Pull request by klukonin**
Maintainer: Ashkan Jazayeri ashkan@jazayeri.net
Compile tested: (x86_64, generic, r7087-7197744cd6)
Run tested: (x86_64, generic, r7087-7197744cd6, af-packet)
Description:

Suricata is a network IDS, IPS and NSM engine. This commit adds a
recent stable version of this package to the packages tree.
Luka Perkov's Snort package was used as a template for writing this one.
Implementations on Selks, Arch, Turis and Debian have also been reviewed.

Non-blocking service reload command taken from:
http://suricata.readthedocs.io/en/latest/rule-management/rule-reload.html

This commit is based on the original @ashkanj PR https://github.com/openwrt/packages/pull/6168
All chanes was squashed into one commit and rebased against master to make the merge possible.

Signed-off-by: Kirill Lukonin <klukonin@gmail.com>

### CODE REVIEW

File: net/suricata/Makefile [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+endif
+ifeq ($(CONFIG_SURICATA_NFLOG),y)
+CONFIGURE_ARGS += \
+	--enable-nflog
+endif
+ifeq ($(CONFIG_SURICATA_NFQ),y)
+CONFIGURE_ARGS += \
+	--enable-nfqueue
+endif
+ifeq ($(CONFIG_SURICATA_PIE),y)
```
Comment (neheb) [inline net/suricata/Makefile]: This should not even be an option. There's PKG_ASLR_PIE that sets this globally.

**Comment by BKPepe** [comment]
Let me share some of my thoughts about it.

- Many comments from the previous PR are not addressed.
- Unfortunately, it seems the maintainer is not active on Github, but I think he wants to have authorship in the commit. Use cherry-pick. However, you or somebody else needs to take maintainer of this package. That can be done in a separate commit.
- There is already available old-stable version 4.0.7 available. 
However, there is release 4.1, which requires Rust according to https://suricata-ids.org/2018/11/06/suricata-4-1-released/
I'd rather see the latest version, but the old stable, for now, is just fine.
- Change the DEPENDS sections in your commit. There are three and it doesn't look good.
You can take a look in some Makefiles, which are here or in this [one](https://github.com/openwrt/packages/blob/master/net/keepalived/Makefile#L61).
- While changing DEPENDS, you can reorder some things in Makefile to be more equal to other Makefiles, which we have and some things like PKG_LICENSE_FILES, PKG_CPE_ID.
- URL of the project is https://suricata-ids.org/

File: net/suricata/Makefile [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+		https://github.com/OISF/suricata/releases/download/$(PKG_NAME)-$(PKG_VERSION)/ \
+		https://www.openinfosecfoundation.org/download/
+
+PKG_HASH:=617e83b6e20b03aa7d5e05a980d3cb6d2810ec18a6f15a36bf66c81c9c0a2abb
+PKG_FIXUP:=autoreconf
+PKG_BUILD_PARALLEL:=1
+PKG_INSTALL:=1
+
+PKG_LICENSE:=GPL-2.0
+PKG_MAINTAINER:=Ashkan Jazayeri <ashkan@jazayeri.net>
```
Comment (neheb) [inline net/suricata/Makefile]: If the previous maintainer has been away for this long, there's little chance he's going to maintain this. I would remove and replace this with someone who will.

**Comment by diizzyy** [comment]
Adding to the comments from others
I'm against importing something that's more or less already deprecated upstream.
This will only run decently on aarch64 and x86_64 hardware so might as well make it a requirement (if even imported)

**Comment by ashkanj** [comment]
Ran a quick test and 4.1.3 compiles without an error. You only need to add a --disable-suricata-update to the configure args.
I'll try to send another PR in upcoming days covering issues discussed above.
However I agree that most of the stuff like rust, Hyperscan and libnss are yet to be covered and we need more feedback from people using this package.

**Comment by klukonin** [comment]
> Ran a quick test and 4.1.3 compiles without an error. You only need to add a --disable-suricata-update to the configure args.
> I'll try to send another PR in upcoming days covering issues discussed above.
> However I agree that most of the stuff like rust, Hyperscan and libnss are yet to be covered and we need more feedback from people using this package.

Should I close this PR and continue the original one?

**Comment by BKPepe** [comment]
Let's close this issue due to lack of activity of submitter and also of the previous submitter. Even I would appreciate having Suricata in OpenWrt, but we can not increase the number of packages, which won't be maintained. 

The other and important reason is that Rust is not supported in OpenWrt, yet.
Since [Suricata 5.0.0](https://suricata-ids.org/2019/04/30/call-for-testing-announcing-suricata-5-0-0-beta1/), Rust is no longer as optional, but mandatory. The first step before Suricata can be accepted to packages repository is to have Rust support (at least as a package) in OpenWrt.

Right now, it doesn't make sense to me to accept this package as the old version will be someday EOL and in the near future we will need to update to the latest versions due to some (security) issues which can appear.

**Comment by klukonin** [comment]
Thank you, Josef.
Sad, but true.

**Pull request closed without merging**